geomesa-filter/src/test/scala/org/locationtech/geomesa/filter/FilterHelperTest.scala

```diff
@@ -41,6 +44,31 @@ class FilterHelperTest extends Specification {
 
   "FilterHelper" should {
 
+    "optimize the inArray function" >> {
+      val filter = ff.function("inArray", ff.literal("foo"), ff.literal(Array("foo", "bar")))
+      val filter2 = ff.equals(
+        ff.function("inArray", ff.property("name"), ff.literal(new util.ArrayList(util.Arrays.asList("foo", "bar")))),
+        ff.literal(java.lang.Boolean.TRUE)
+      )
+
+      val sf = new ScalaSimpleFeature(sft, "1",
+        Array(new Date(), new Integer(1), new Integer(2), new Integer(3), new Integer(4), WKTUtils.read("POINT(1 5)"), "foo"))
+      val filter3 = updateFilter(filter2)
+
+      filter3.isInstanceOf[OrImpl] mustEqual(true)
+
+      val filter4 = ff.equals(
+        ff.literal(java.lang.Boolean.TRUE),
+        ff.function("inArray",
+          ff.property("name"),
+          ff.literal(new util.ArrayList(util.Arrays.asList("foo", "bar"))))
+      )
+      val filter5 = updateFilter(filter4)
+
+      filter5.isInstanceOf[OrImpl] mustEqual(true)
+      ok
```

Comment:
would it also be possible to show an ECQL example? or is that not relevant? I forget now, would a user ever need to invoke this themselves through e.g. geoserver where it would have to be ECQL?

Reply:
(To my knowledge) There's no ECQL support for arrays.  

ECQL.toFilter will print out something like...
```inArray(name,'[foo, bar]') = true```

If you run that through ECQL.toCQL, the [foo, bar] is turned into a literal, not an (Array)List.

I think the ECQL 'support' for this query pattern is the "optimized" version that we are creating with the Visitor.